src/httpserver.cpp

```diff
@@ -443,8 +443,15 @@ void InterruptHTTPServer()
         // Reject requests on current connections
         evhttp_set_gencb(eventHTTP, http_reject_request_cb, nullptr);
     }
-    if (workQueue)
-        workQueue->Interrupt();
+    if (workQueue) {
+        if (eventBase) {
+            event_base_once(eventBase, -1, EV_TIMEOUT, [](evutil_socket_t, short, void*) {
+                 workQueue->Interrupt();
+            }, nullptr, nullptr);
+        } else {
+            workQueue->Interrupt();
```

Comment:
`workQueue` is not created if `eventBase` couldn't be created, so it seems this `else {}` clause is redundant?

Reply:
@laanwj changed approach, please take a new look.